doc/upload.md

```diff
@@ -0,0 +1,153 @@
+
+For Minikube, you can exexute the following which will create a NodePort service accessible from your host:
+
+```bash
+cat <<EOF | kubectl apply -f -
+kind: Service
+metadata:
+  name: cdi-uploadproxy-nodeport
+  namespace: kube-system
+  labels:
+    cdi.kubevirt.io: "cdi-uploadproxy"
+spec:
+  type: NodePort
+  ports:
+    - port: 443
+      targetPort: 8443
+      nodePort: 31001
+      protocol: TCP
+  selector:
+    cdi.kubevirt.io: cdi-uploadproxy
+EOF
+```
+
+## Create a PersistentVolumeClaim
+Annotating a PVC with `cdi.kubevirt.io/storage.upload.target` marks the PVC as a target for CDI Upload.
+
+Take a look at at `manifests/example/upload-pvc.yaml` for an example.
+```
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: upload-test
+  labels:
+    app: containerized-data-importer
+  annotations:
+    cdi.kubevirt.io/storage.upload.target: ""
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+
+```
+```
+kubectl apply -f manifests/example/upload-pvc.yaml
+```
+
+## Request an Upload Token
+Before sending data to the Upload Proxy, and Upload Token must be requested.  The CDI API Server validatees that the user has permissions to `post` to `uploadtokenrequest` resources.
+
+Take a look at at `manifests/example/upload-token.yaml` for an example.
+```
+apiVersion: upload.cdi.kubevirt.io/v1alpha1
+kind: UploadTokenRequest
+metadata:
+  name: upload-test
+  namespace: default
+spec:
+  pvcName: upload-test
+
+```
+```bash
+kubectl apply -f manifests/example/upload-token.yaml -o yaml
+apiVersion: upload.cdi.kubevirt.io/v1alpha1
+kind: UploadTokenRequest
+metadata:
+  annotations:
+    kubectl.kubernetes.io/last-applied-configuration: |
+      {"apiVersion":"upload.cdi.kubevirt.io/v1alpha1","kind":"UploadTokenRequest","metadata":{"annotations":{},"name":"upload-test","namespace":"default"},"spec":{"pvcName":"upload-test"}}
+  creationTimestamp: null
+  name: upload-test
+  namespace: default
+spec:
+  pvcName: upload-test
+status:
+  token: eyJhbGciOiJQUzUxMiIsImtpZCI6IiJ9.eyJwdmNOYW1lIjoidXBsb2FkLXRlc3QiLCJuYW1lc3BhY2UiOiJkZWZhdWx0IiwiY3JlYXRpb25UaW1lc3RhbXAiOiIyMDE4LTA5LTIxVDE4OjEyOjE5LjQwODI1MDQ4NFoifQ.JWk1VyvzSse3eFiBROKgGoLnOPCiYW9JdDWKXFROEL6XY0O5lFb1R0rwdfWwC3BBOtEA9mC9x3ZGYPnYWO-5G_r1fWKHjF-zifrCX_3Dhp3vfSq6Zfpu-vV0Qn0A3YkSCCmiC_nONAhVjEDuQsRFIKwYcxBoEOpye92ggH2u5FxQE7FwxxH6-RHun9tc_lIFX-ZFKnq7n5tWbjsTmAZI_4rDNgYkVFhFtENU6e-5_Ncokxs3YVzkbSrXweZpRmmaYQOmZhjXSLjKED_2FVq7tYeVueEEhKC_zJ-AEivstALPwPjiwyWXJyfE3dCmbA1sBKuNUrAaDlBvSAp1uPV9eQ
+  ```
+  Save the `token` field of the response status.  It will be used to authorize our CDI Upload request. Tokens are good for 5 minutes.
+
+If you have [jq](https://github.com/stedolan/jq) installed, you can capture the token in an environment variable by doing something like this:
+```bash
+TOKEN=`kubectl apply -f manifests/example/upload-token.yaml -o json | jq -r '.status.token'`
```

Comment:
You can do -o jsonpath and obtain the same result without jq (in fact jsonpath uses jq in the back ground)

Reply:
Whoa!  I did not know that!  Will update accordingly!